compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1639,8 +1639,10 @@ private int getObjectFuncCount(BObjectTypeSymbol sym) {
     }
 
     public boolean checkRecordEquivalency(BRecordType rhsType, BRecordType lhsType, Set<TypePair> unresolvedTypes) {
-        // If the LHS record is closed and the RHS record is open, the records aren't equivalent
-        if (lhsType.sealed && !rhsType.sealed) {
+        // If the LHS record is closed and the RHS record is open and the rest field type of RHS is not a 'never'
+        // type, the records aren't equivalent
+        if (lhsType.sealed && !rhsType.sealed &&
+                (rhsType.restFieldType == null || rhsType.restFieldType.tag != TypeTags.NEVER)) {
```

Comment:
IINM, rest field type is never null. It's usually noType. I think we can get rid of the null check.